smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -70,6 +57,19 @@ contract MixinPurchase is
       toKey.expirationTimestamp = block.timestamp + expirationDuration;
     }
 
+    // Let's get the actual price for the key from the Unlock smart contract
+    uint discount;
+    uint tokens;
+    uint inMemoryKeyPrice = keyPrice;
+    (discount, tokens) = unlockProtocol.computeAvailableDiscountFor(_recipient, inMemoryKeyPrice);
```

Comment:
So, aren't we missing information here? The currency itself will need to be known by `unlockProtocol` to decide what amount of discount it needs to grant? [we may have to "whitelist" the ERC20 which give right to discounts]